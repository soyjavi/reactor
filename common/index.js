import { LAYOUT } from './layout/index';
import SHAPE from './shape';
import THEME from './theme';

export { LAYOUT, SHAPE, THEME };
