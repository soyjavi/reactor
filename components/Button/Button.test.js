import React from 'react';
import renderer from 'react-test-renderer';

import { THEME } from '../../common';
import { Button, Price } from '..';

jest.useFakeTimers();

describe('<Button>', () => {
  it('renders', () => {
    const snapshot = renderer.create(<Button />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  it('when {title}', () => {
    const snapshot = renderer.create(<Button title="Press me" />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  it('when {activity}', () => {
    const snapshot = renderer.create(<Button activity title="Press me" />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  it('when {borderRadius}', () => {
    const snapshot = renderer.create(<Button borderRadius={10} title="Press me" />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  it('when {children}', () => {
    const snapshot = renderer
      .create(
        <Button>
          <Price headline color="white" value={19.95} symbol="$" />
        </Button>,
      )
      .toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  it('when {color}', () => {
    const snapshot = renderer.create(<Button color={THEME.COLOR.BLACK} title="Press me" />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  it('when {disabled}', () => {
    const snapshot = renderer.create(<Button disabled title="Press me" />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  it('when {icon}', () => {
    const snapshot = renderer.create(<Button icon="close" />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  it('when {icon} & {title}', () => {
    const snapshot = renderer.create(<Button icon="close" title="Press me" />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  it('when {onPress}', () => {
    // @TODO: Test <Button /> {onPress}
  });

  it('when {outlined}', () => {
    const snapshot = renderer.create(<Button outlined title="Press me" />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  it('when {size}', () => {
    const snapshot = renderer.create(<Button size="S" title="Press me" />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  it('when {wide}', () => {
    const snapshot = renderer.create(<Button wide title="Press me" />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  it('when {style}', () => {
    const snapshot = renderer
      .create(<Button title="Press me" style={{ backgroundColor: 'rgba(0,255,0,0.25)' }} />)
      .toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  it('hook:styler', () => {
    const snapshot = renderer.create(<Button title="Press me" marginBottom="M" />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });
});
