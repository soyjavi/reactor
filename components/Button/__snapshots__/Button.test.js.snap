// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button> renders 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  pointerEvents="none"
  style={
    Array [
      undefined,
      Array [
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "alignSelf": "flex-start",
          "justifyContent": "center",
          "width": "auto",
        },
        undefined,
        Object {
          "backgroundColor": "#ff8c5f",
          "borderRadius": undefined,
        },
        undefined,
        Object {
          "height": 48,
          "paddingHorizontal": 16,
        },
        undefined,
      ],
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
      }
    }
  />
  <View
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderRadius": 0,
          "bottom": 0,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#ffffff",
          "height": 0,
          "opacity": 0,
          "width": 0,
          "zIndex": 0,
        }
      }
    />
  </View>
</View>
`;

exports[`<Button> when {activity} 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  pointerEvents="none"
  style={
    Array [
      undefined,
      Array [
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "alignSelf": "flex-start",
          "justifyContent": "center",
          "width": "auto",
        },
        undefined,
        Object {
          "backgroundColor": "#ff8c5f",
          "borderRadius": undefined,
        },
        undefined,
        Object {
          "height": 48,
          "paddingHorizontal": 16,
        },
        undefined,
      ],
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#ffffff",
          "height": 8,
          "marginHorizontal": 4,
          "opacity": 0,
          "width": 8,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "#ffffff",
          "height": 8,
          "marginHorizontal": 4,
          "opacity": 0,
          "width": 8,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "#ffffff",
          "height": 8,
          "marginHorizontal": 4,
          "opacity": 0,
          "width": 8,
        }
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderRadius": 0,
          "bottom": 0,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#ffffff",
          "height": 0,
          "opacity": 0,
          "width": 0,
          "zIndex": 0,
        }
      }
    />
  </View>
</View>
`;

exports[`<Button> when {borderRadius} 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  pointerEvents="none"
  style={
    Array [
      undefined,
      Array [
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "alignSelf": "flex-start",
          "justifyContent": "center",
          "width": "auto",
        },
        undefined,
        Object {
          "backgroundColor": "#ff8c5f",
          "borderRadius": 10,
        },
        undefined,
        Object {
          "height": 48,
          "paddingHorizontal": 16,
        },
        undefined,
      ],
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": undefined,
            "textAlignVertical": "center",
          },
          Object {
            "fontSize": 16,
            "lineHeight": 24,
          },
          undefined,
          false,
          Object {
            "color": "#ffffff",
          },
          Array [
            Object {
              "fontFamily": undefined,
              "fontWeight": "700",
              "paddingHorizontal": 8,
              "textAlignVertical": "center",
            },
            false,
          ],
        ]
      }
    >
      Press me
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderRadius": 0,
          "bottom": 0,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "borderRadius": 10,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#ffffff",
          "height": 0,
          "opacity": 0,
          "width": 0,
          "zIndex": 0,
        }
      }
    />
  </View>
</View>
`;

exports[`<Button> when {children} 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  pointerEvents="none"
  style={
    Array [
      undefined,
      Array [
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "alignSelf": "flex-start",
          "justifyContent": "center",
          "width": "auto",
        },
        undefined,
        Object {
          "backgroundColor": "#ff8c5f",
          "borderRadius": undefined,
        },
        undefined,
        Object {
          "height": 48,
          "paddingHorizontal": 16,
        },
        undefined,
      ],
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": undefined,
            "textAlignVertical": "center",
          },
          Object {
            "fontSize": 32,
            "fontWeight": "700",
          },
          undefined,
          false,
          Object {
            "color": "white",
          },
          undefined,
        ]
      }
    >
      $19,95
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderRadius": 0,
          "bottom": 0,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#ffffff",
          "height": 0,
          "opacity": 0,
          "width": 0,
          "zIndex": 0,
        }
      }
    />
  </View>
</View>
`;

exports[`<Button> when {color} 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  pointerEvents="none"
  style={
    Array [
      undefined,
      Array [
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "alignSelf": "flex-start",
          "justifyContent": "center",
          "width": "auto",
        },
        undefined,
        Object {
          "backgroundColor": "#191919",
          "borderRadius": undefined,
        },
        undefined,
        Object {
          "height": 48,
          "paddingHorizontal": 16,
        },
        undefined,
      ],
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": undefined,
            "textAlignVertical": "center",
          },
          Object {
            "fontSize": 16,
            "lineHeight": 24,
          },
          undefined,
          false,
          Object {
            "color": "#ffffff",
          },
          Array [
            Object {
              "fontFamily": undefined,
              "fontWeight": "700",
              "paddingHorizontal": 8,
              "textAlignVertical": "center",
            },
            false,
          ],
        ]
      }
    >
      Press me
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderRadius": 0,
          "bottom": 0,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#ffffff",
          "height": 0,
          "opacity": 0,
          "width": 0,
          "zIndex": 0,
        }
      }
    />
  </View>
</View>
`;

exports[`<Button> when {disabled} 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  pointerEvents="none"
  style={
    Array [
      undefined,
      Array [
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "alignSelf": "flex-start",
          "justifyContent": "center",
          "width": "auto",
        },
        undefined,
        Object {
          "backgroundColor": "#d2d2d2",
        },
        undefined,
        Object {
          "height": 48,
          "paddingHorizontal": 16,
        },
        undefined,
      ],
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": undefined,
            "textAlignVertical": "center",
          },
          Object {
            "fontSize": 16,
            "lineHeight": 24,
          },
          undefined,
          false,
          Object {
            "color": "#a3a3a3",
          },
          Array [
            Object {
              "fontFamily": undefined,
              "fontWeight": "700",
              "paddingHorizontal": 8,
              "textAlignVertical": "center",
            },
            false,
          ],
        ]
      }
    >
      Press me
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderRadius": 0,
          "bottom": 0,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#a3a3a3",
          "height": 0,
          "opacity": 0,
          "width": 0,
          "zIndex": 0,
        }
      }
    />
  </View>
</View>
`;

exports[`<Button> when {icon} & {title} 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  pointerEvents="none"
  style={
    Array [
      undefined,
      Array [
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "alignSelf": "flex-start",
          "justifyContent": "center",
          "width": "auto",
        },
        undefined,
        Object {
          "backgroundColor": "#ff8c5f",
          "borderRadius": undefined,
        },
        undefined,
        Object {
          "height": 48,
          "paddingHorizontal": 16,
        },
        undefined,
      ],
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
      }
    }
  >
    <Text />
    <Text
      style={
        Array [
          Object {
            "fontFamily": undefined,
            "textAlignVertical": "center",
          },
          Object {
            "fontSize": 16,
            "lineHeight": 24,
          },
          undefined,
          false,
          Object {
            "color": "#ffffff",
          },
          Array [
            Object {
              "fontFamily": undefined,
              "fontWeight": "700",
              "paddingHorizontal": 8,
              "textAlignVertical": "center",
            },
            false,
          ],
        ]
      }
    >
      Press me
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderRadius": 0,
          "bottom": 0,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#ffffff",
          "height": 0,
          "opacity": 0,
          "width": 0,
          "zIndex": 0,
        }
      }
    />
  </View>
</View>
`;

exports[`<Button> when {icon} 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  pointerEvents="none"
  style={
    Array [
      undefined,
      Array [
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "alignSelf": "flex-start",
          "justifyContent": "center",
          "width": "auto",
        },
        undefined,
        Object {
          "backgroundColor": "#ff8c5f",
          "borderRadius": undefined,
        },
        undefined,
        Object {
          "height": 48,
          "paddingHorizontal": 16,
        },
        undefined,
      ],
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
      }
    }
  >
    <Text />
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderRadius": 0,
          "bottom": 0,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#ffffff",
          "height": 0,
          "opacity": 0,
          "width": 0,
          "zIndex": 0,
        }
      }
    />
  </View>
</View>
`;

exports[`<Button> when {outlined} 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  pointerEvents="none"
  style={
    Array [
      undefined,
      Array [
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "alignSelf": "flex-start",
          "justifyContent": "center",
          "width": "auto",
        },
        Object {
          "borderWidth": 1,
        },
        Object {
          "borderColor": "#ff8c5f",
          "borderRadius": undefined,
        },
        undefined,
        Object {
          "height": 48,
          "paddingHorizontal": 16,
        },
        undefined,
      ],
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": undefined,
            "textAlignVertical": "center",
          },
          Object {
            "fontSize": 16,
            "lineHeight": 24,
          },
          undefined,
          false,
          Object {
            "color": "#ff8c5f",
          },
          Array [
            Object {
              "fontFamily": undefined,
              "fontWeight": "700",
              "paddingHorizontal": 8,
              "textAlignVertical": "center",
            },
            false,
          ],
        ]
      }
    >
      Press me
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderRadius": 0,
          "bottom": 0,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#ff8c5f",
          "height": 0,
          "opacity": 0,
          "width": 0,
          "zIndex": 0,
        }
      }
    />
  </View>
</View>
`;

exports[`<Button> when {size} 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  pointerEvents="none"
  style={
    Array [
      undefined,
      Array [
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "alignSelf": "flex-start",
          "justifyContent": "center",
          "width": "auto",
        },
        undefined,
        Object {
          "backgroundColor": "#ff8c5f",
          "borderRadius": undefined,
        },
        undefined,
        Object {
          "height": 32,
          "paddingHorizontal": 8,
        },
        undefined,
      ],
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": undefined,
            "textAlignVertical": "center",
          },
          Object {
            "fontSize": 16,
            "lineHeight": 24,
          },
          undefined,
          false,
          Object {
            "color": "#ffffff",
          },
          Array [
            Object {
              "fontFamily": undefined,
              "fontWeight": "700",
              "paddingHorizontal": 8,
              "textAlignVertical": "center",
            },
            Object {
              "fontSize": 12,
            },
          ],
        ]
      }
    >
      Press me
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderRadius": 0,
          "bottom": 0,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#ffffff",
          "height": 0,
          "opacity": 0,
          "width": 0,
          "zIndex": 0,
        }
      }
    />
  </View>
</View>
`;

exports[`<Button> when {style} 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  pointerEvents="none"
  style={
    Array [
      undefined,
      Array [
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "alignSelf": "flex-start",
          "justifyContent": "center",
          "width": "auto",
        },
        undefined,
        Object {
          "backgroundColor": "#ff8c5f",
          "borderRadius": undefined,
        },
        undefined,
        Object {
          "height": 48,
          "paddingHorizontal": 16,
        },
        Object {
          "backgroundColor": "rgba(0,255,0,0.25)",
        },
      ],
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": undefined,
            "textAlignVertical": "center",
          },
          Object {
            "fontSize": 16,
            "lineHeight": 24,
          },
          undefined,
          false,
          Object {
            "color": "#ffffff",
          },
          Array [
            Object {
              "fontFamily": undefined,
              "fontWeight": "700",
              "paddingHorizontal": 8,
              "textAlignVertical": "center",
            },
            false,
          ],
        ]
      }
    >
      Press me
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderRadius": 0,
          "bottom": 0,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#ffffff",
          "height": 0,
          "opacity": 0,
          "width": 0,
          "zIndex": 0,
        }
      }
    />
  </View>
</View>
`;

exports[`<Button> when {title} 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  pointerEvents="none"
  style={
    Array [
      undefined,
      Array [
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "alignSelf": "flex-start",
          "justifyContent": "center",
          "width": "auto",
        },
        undefined,
        Object {
          "backgroundColor": "#ff8c5f",
          "borderRadius": undefined,
        },
        undefined,
        Object {
          "height": 48,
          "paddingHorizontal": 16,
        },
        undefined,
      ],
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": undefined,
            "textAlignVertical": "center",
          },
          Object {
            "fontSize": 16,
            "lineHeight": 24,
          },
          undefined,
          false,
          Object {
            "color": "#ffffff",
          },
          Array [
            Object {
              "fontFamily": undefined,
              "fontWeight": "700",
              "paddingHorizontal": 8,
              "textAlignVertical": "center",
            },
            false,
          ],
        ]
      }
    >
      Press me
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderRadius": 0,
          "bottom": 0,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#ffffff",
          "height": 0,
          "opacity": 0,
          "width": 0,
          "zIndex": 0,
        }
      }
    />
  </View>
</View>
`;

exports[`<Button> when {wide} 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  pointerEvents="none"
  style={
    Array [
      undefined,
      Array [
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "alignSelf": "flex-start",
          "justifyContent": "center",
          "width": "auto",
        },
        undefined,
        Object {
          "backgroundColor": "#ff8c5f",
          "borderRadius": undefined,
        },
        Object {
          "width": "100%",
        },
        Object {
          "height": 48,
          "paddingHorizontal": 16,
        },
        undefined,
      ],
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": undefined,
            "textAlignVertical": "center",
          },
          Object {
            "fontSize": 16,
            "lineHeight": 24,
          },
          undefined,
          false,
          Object {
            "color": "#ffffff",
          },
          Array [
            Object {
              "fontFamily": undefined,
              "fontWeight": "700",
              "paddingHorizontal": 8,
              "textAlignVertical": "center",
            },
            false,
          ],
        ]
      }
    >
      Press me
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderRadius": 0,
          "bottom": 0,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#ffffff",
          "height": 0,
          "opacity": 0,
          "width": 0,
          "zIndex": 0,
        }
      }
    />
  </View>
</View>
`;
